tasklist/client/src/Tasks/Task/Details/Header.tsx

```diff
@@ -139,15 +140,47 @@ const AssignButton: React.FC<{
     } catch (error) {
       const errorMessage = (error as Error).message ?? '';
 
-      setAssignmentStatus('off');
+      if (shouldDisplayInfoNotification(errorMessage)) {
+        tracking.track({
+          eventName: isAssigned
+            ? 'task-unassignment-delayed-notification'
+            : 'task-assignment-delayed-notification',
+        });
+        notificationsStore.displayNotification({
+          kind: 'info',
+          title: isAssigned
+            ? t('taskDetailsUnassignmentDelayInfoTitle')
+            : t('taskDetailsAssignmentDelayInfoTitle'),
+          subtitle: isAssigned
+            ? t('taskDetailsUnassignmentDelayInfoSubtitle')
+            : t('taskDetailsAssignmentDelayInfoSubtitle'),
+          isDismissable: true,
+        });
+        return;
+      }
 
-      if (shouldDisplayNotification(errorMessage)) {
+      const statusMatch = errorMessage.match(/status:\s*(\d+)/);
+      const statusCode = statusMatch ? parseInt(statusMatch[1], 10) : undefined;
+
+      setAssignmentStatus('off');
+      if (shouldDisplayNotification(errorMessage, statusCode)) {
+        if (statusCode === 409) {
+          tracking.track({
+            eventName: isAssigned
+              ? 'task-unassignment-rejected-notification'
+              : 'task-assignment-rejected-notification',
+          });
+        }
```

Comment:
Parsing these raw responses is not good solution, we did this in the past as a temporary solution  as you seen it became permanent because it was never prioritized.

I understand that the V1 API will be deprecated in the coming quarters so it doesn't make sense to have proper error codes for the entire API but we could at least some basic error codes for this specific endpoint would be a good middle.
Parsing Java error like on `shouldDisplayInfoNotification` is not very nice

Reply:
I fully understand and agree with the point raised. I was also hesitant to rely on raw responses, but I wasn’t sure how much effort was appropriate given the V1 API deprecation. I’ve now added specific error codes for this endpoint as a middle ground.